### PR TITLE
Fix automation workflow 

### PR DESCRIPTION
**Pull request by sonalivdeshpande**

### CODE REVIEW

File: .github/workflows/merge-commits-to-next-branch.yml [hunk 1]
```diff
@@ -18,8 +17,9 @@ jobs:
         with:
           node-version: 14
       - run: npm ci
```
Comment (tylerbutler) [inline .github/workflows/merge-commits-to-next-branch.yml:19]: Is this needed? It's really slow -- what's it needed for?

File: .github/workflows/merge-commits-to-next-branch.yml [hunk 2]
```diff
@@ -18,8 +17,9 @@ jobs:
...
-      - run: npm install -g @azure/storage-queue
+      - run: npm install @azure/storage-queue
       - uses: actions/github-script@7a5c598405937d486b0331594b5da2b14db670da # pin@v6
+        id: deque-message
```
Comment (tylerbutler) [inline .github/workflows/merge-commits-to-next-branch.yml:22]:
```suggestion
        id: dequeue-message
```

File: .github/workflows/merge-commits-to-next-branch.yml [hunk 3]
```diff
@@ -29,72 +29,73 @@ jobs:
             const queueClient = queueServiceClient.getQueueClient(queueName);
             var receivedMessages = await queueClient.receiveMessages();
             const message = receivedMessages.receivedMessageItems[0];
-            $COMMIT_INFO = JSON.parse(message.messageText);
+            const firstMessage = JSON.parse(message.messageText);
+            return firstMessage;
+    outputs:
+      LABEL: ${{ fromJSON(steps.deque-message.outputs.result).label }}
+      AUTHOR: ${{ fromJSON(steps.deque-message.outputs.result).author }}
+      SHA: ${{ fromJSON(steps.deque-message.outputs.result).sha }}
```
Comment (tylerbutler) [inline .github/workflows/merge-commits-to-next-branch.yml:37]:
```suggestion
      LABEL: ${{ fromJSON(steps.dequeue-message.outputs.result).label }}
      AUTHOR: ${{ fromJSON(steps.dequeue-message.outputs.result).author }}
      SHA: ${{ fromJSON(steps.dequeue-message.outputs.result).sha }}
```

File: .github/workflows/merge-commits-to-next-branch.yml [hunk 4]
```diff
@@ -29,72 +29,73 @@ jobs:
...
   create-branch:
     runs-on: ubuntu-latest
-    if: ( ($COMMIT_INFO).label == 'queued' )
+    permissions:
+      contents: write
+      packages: write
```
Comment (tylerbutler) [inline .github/workflows/merge-commits-to-next-branch.yml:42]: try

```suggestion
    permissions:
      actions: write
      checks: write
      contents: write
      issues: write
      pull-requests: write
      statuses: write
```

And if that doesn't work try `permissions: write-all` just for the job that pushes the HEAD.

File: .github/workflows/merge-commits-to-next-branch.yml [hunk 5]
```diff
@@ -29,72 +29,73 @@ jobs:
...
   createPullRequest:
     runs-on: ubuntu-latest
-    needs: [ create-branch ]
-    if: ( ($COMMIT_INFO).label == 'queued' && needs.create-branch.result == 'success'
-      )
+    needs: [ create-branch, dequeue ]
+    if: needs.create-branch.result == 'success'
     steps:
-      - name: Create Pull Request
+      - run: npm install @octokit/core
```
Comment (tylerbutler) [inline .github/workflows/merge-commits-to-next-branch.yml:70]: can you use a global install? This is basically installing the package as a dependency of the main repo, which will be slow and unnecessary.

File: .github/workflows/merge-commits-to-next-branch.yml [hunk 6]
```diff
@@ -29,72 +29,73 @@ jobs:
...
     runs-on: ubuntu-latest
-    needs: [ createPullRequest ]
+    needs: [ createPullRequest, dequeue ]
     if: ( needs.createPullRequest.result == 'success' )
     steps:
       - uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # pin@v2
       - uses: actions/setup-node@1f8c6b94b26d0feae1e387ca63ccbdc44d27b561 # pin@v2
         with:
           node-version: 14
       - run: npm ci
```
Comment (tylerbutler) [inline .github/workflows/merge-commits-to-next-branch.yml:97]: Same as above.

File: .github/workflows/push-to-azure-queue.yml [hunk 7]
```diff
@@ -35,8 +36,18 @@ jobs:
...
+            const { Octokit } = require("@octokit/core");
+            const pr = ${{ github.event.pull_request.number }}
+            const token = `${{ secrets.GITHUB_TOKEN }}`;
+            const octokit = new Octokit({ auth: token })
+            octokit.rest.pulls.createReviewComment({
+              owner: 'sonalivdeshpande',
+              repo: 'FluidFramework',
+              pull_number: pr,
+              body: 'This commit is queued for merging with the `test_next` branch! Please ignore this PR for now. Contact @microsoft/fluid-cr-infra for help.',
+            });
```
Comment (tylerbutler) [inline .github/workflows/push-to-azure-queue.yml:53]: This is OK, but consider using [this action](https://github.com/marocchino/sticky-pull-request-comment) that we already use elsewhere. It's quite nice.

https://github.com/microsoft/FluidFramework/blob/11848cd181cad0729b656cf9e3e5a9fe11816942/.github/workflows/linkcheck-reporter.yml#L24-L30

That example uses a file as input, but there are other simple examples in the action docs.

File: .github/workflows/push-to-azure-queue.yml [hunk 8]
```diff
@@ -35,8 +36,18 @@ jobs:
...
-          body: |
-            'This commit is queued for merging with the `test_next` branch! Please ignore this PR for now. Contact @microsoft/fluid-cr-infra for help.'
+          token: ${{ secrets.WORKFLOW_TOKEN }}
+          script: |
+            const { Octokit } = require("@octokit/core");
+            const pr = ${{ github.event.pull_request.number }}
+            const token = `${{ secrets.GITHUB_TOKEN }}`;
+            const octokit = new Octokit({ auth: token })
+            octokit.rest.pulls.createReviewComment({
+              owner: 'sonalivdeshpande',
```
Comment (tylerbutler) [inline .github/workflows/push-to-azure-queue.yml:49]: This needs to be configurable or come from the context of the workflow itself (maybe `${{ github.repository_owner }}`?

File: .github/workflows/merge-commits-to-next-branch.yml [hunk 9]
```diff
@@ -29,72 +29,73 @@ jobs:
...
             const message = receivedMessages.receivedMessageItems[0];
-            $COMMIT_INFO = JSON.parse(message.messageText);
+            const firstMessage = JSON.parse(message.messageText);
+            return firstMessage;
+    outputs:
+      LABEL: ${{ fromJSON(steps.deque-message.outputs.result).label }}
+      AUTHOR: ${{ fromJSON(steps.deque-message.outputs.result).author }}
+      SHA: ${{ fromJSON(steps.deque-message.outputs.result).sha }}
   create-branch:
     runs-on: ubuntu-latest
```
Comment (tylerbutler) [inline .github/workflows/merge-commits-to-next-branch.yml:39]:
```suggestion
    runs-on: ubuntu-latest
    if: github.repository == 'microsoft/FluidFramework'
```

Adding this for all the jobs (or the top level ones) should prevent it from running outside the main repo.

**Review by tylerbutler — APPROVED** [review]

File: .github/workflows/merge-commits-to-next-branch.yml (hunk 4 above)
Comment (sonalivdeshpande) [reply .github/workflows/merge-commits-to-next-branch.yml:42]: write-all didn't work on my fork. I'll try adding all permissions

File: .github/workflows/merge-commits-to-next-branch.yml (hunk 5 above)
Comment (sonalivdeshpande) [reply .github/workflows/merge-commits-to-next-branch.yml]: testing this as part of a follow-up PR

File: .github/workflows/push-queue.yml [hunk 10]
```diff
@@ -35,8 +34,8 @@ jobs:
     runs-on: ubuntu-latest
     needs: [ queue-pr ]
     steps:
-      - name: Create commit comment
-        uses: peter-evans/commit-comment@3f3629d9da4bf387b09fc75456a5b8a344ba5641 # pin@v2
+      - name: Post comment on pr
+        uses: marocchino/sticky-pull-request-comment@v2
```
Comment (tylerbutler) [inline .github/workflows/push-queue.yml:38]:
```suggestion
        uses: marocchino/sticky-pull-request-comment@39c5b5dc7717447d0cba270cd115037d32d28443 # pin@v2
```

**Review by tylerbutler — APPROVED** [review]

**Pull request merged**